playbooks/operationalizing/secure-forward-splunk-container.adoc

```diff
@@ -0,0 +1,269 @@
+
+* Port: 9997
+* Source name override: OCP
+* Source Type: _json
+
+NOTE: It is beyond the scope of this document for the creation of a custom link:https://docs.splunk.com/Splexicon%3ASourcetype[sourcetype] to perform advanced parsing of the data received.
+
+Complete the necessary fields to finalize the addition of the new TCP input
+
+NOTE: Open applicable firewall ports on the Splunk machine to allow for data to be received
+
+Since the fluent-plugin-splunk-ex plugin sends data to Splunk in batches, Splunk must be configured to separate messages based on the presence of a new line character.
+
+Configure the _/opt/splunk/apps/search/local/props.conf_ file with the following content which will configure messages received from the _OCP_ source to split messages correctly:
+
+[source]
+----
+[source::OCP]
+SHOULD_LINEMERGE = false
+----
+
+Restart Splunk to apply the changes:
+
+[source]
+----
+systemctl restart splunk
+----
+
+
+=== Containerized Fluentd
+
+A standalone instance of Fluentd serves as a broker between the OpenShift aggregated logging solution and Splunk. It receives messages sent by the secure forward plugin and forwards them to Splunk using the link:https://github.com/gtrevg/fluent-plugin-splunk-ex[fluent-plugin-splunk-ex] plugin.
+In this implementation, the "external" fluentd instance will be run in OpenShift.
+
+==== Source
+The source code for the implementation can be found link:https://github.com/themoosman/openshift-fluentd-forwarder[here].
+
+==== Create Fluentd Build Configuration
+
+Create the fluentd-forwarder build configuration in the logging project.
+
+[source]
+----
+oc project logging && \
+oc process -f fluentd-forwarder-build-config-template.yaml \
+-p GIT_URI=https://github.com/themoosman/openshift-fluentd-forwarder.git | oc create -f
+----
+
+Build the Docker image in OpenShift and monitor for any errors.
+
+[source]
+----
+oc start-build fluentd-forwarder-build --follow=true
+----
+
+
+==== Create Fluentd Deployment Configuration
+
+Create the fluentd-forwarder deployment configuration in the logging project.
+
+[source]
+----
+oc project logging && \
+oc process -f fluentd-forwarder-template.yaml \
+-p "P_TARGET_TYPE=splunk_ex" \
+-p "P_TARGET_HOST=10.10.10.10" \ <1>
+-p "P_TARGET_PORT=9997" \ <2>
+-p "P_SHARED_KEY=ocpaggregatedloggingsharedkey" <3>
+-p "P_ADDITIONAL_OPTS=output_format json" | <4>
+oc create -f -
+----
+<1> IP address of the Splunk server.
+<2> Port of the Splunk server.
+<3> A shared value between the sender and the receiver.
+<4> Set fluentd to format all logs sent to Splunk in a JSON format.
+
+Validate that the fluentd-forwarder deploys successfully.
+
+==== Update logging-fluentd ConfigMap
+
+The `logging-fluentd` configmap's `data.secure-forward.conf` key needs to be edited as well.
+
+[source]
+----
+oc edit configmap -n logging logging-fluentd
+----
+
+Edit the following YAML:
+
+[source]
+----
+data:
+  secure-forward.conf: |
+    <store>
```

Comment:
I did not need the `<store>` tags. These were added to work around a prior issue. Not sure they are needed now